hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseActionExecutor.java

```diff
@@ -62,6 +64,12 @@ public BaseActionExecutor(HoodieEngineContext context, HoodieWriteConfig config,
    * @param metadata commit metadata of interest.
    */
   protected final void writeTableMetadata(HoodieCommitMetadata metadata, HoodieData<WriteStatus> writeStatus, String actionType) {
+    // Recreate MDT for insert_overwrite_table operation.
+    if (WriteOperationType.INSERT_OVERWRITE_TABLE == metadata.getOperationType()) {
+      HoodieTableMetadataUtil.deleteMetadataTable(table.getMetaClient(), table.getContext(), false);
```

Comment:
can you check if MDT is available and then trigger a delete?

Reply:
Done.